app/src/main/java/org/stepic/droid/ui/activities/LoginActivity.kt

```diff
@@ -166,28 +187,38 @@ class LoginActivity : FragmentActivityBase(), LoginView {
 
     override fun finish() {
         super.finish()
-        overridePendingTransition(R.anim.no_transition, R.anim.slide_out_to_bottom)
+        overridePendingTransition(R.anim.slide_in_from_start, R.anim.slide_out_to_end)
     }
 
     private fun tryLogin() {
-        val login = loginText.text.toString()
-        val password = passwordEditText.text.toString()
+        val login = loginField.text.toString()
+        val password = passwordField.text.toString()
 
         loginPresenter.login(login, password)
     }
 
     override fun onDestroy() {
         loginPresenter.detachView(this)
-        actionbarCloseButtonLayout.setOnClickListener(null)
+        signInWithSocial.setOnClickListener(null)
         loginButton.setOnClickListener(null)
+        launchSignUpButton.setOnClickListener(null)
         if (isFinishing) {
             App.componentManager().releaseLoginComponent(TAG)
         }
         super.onDestroy()
     }
 
+    private fun onClearLoginError() {
+        loginButton.isEnabled = true
+        loginForm.isEnabled = true
+        loginErrorMessage.visibility = View.GONE
+    }
+
     override fun onFailLogin(type: LoginFailType, credential: Credential?) {
-        Toast.makeText(this, getMessageFor(type), Toast.LENGTH_SHORT).show()
+        loginForm.isEnabled = false
```

Comment:
user can't resubmit login+password, when internet connection is lost. It is not right behaviour